hail/src/main/scala/is/hail/expr/ir/functions/Functions.scala

```diff
@@ -251,34 +251,34 @@ abstract class RegistryFunctions {
         srvb.offset)
   }
 
-  def registerCode(mname: String, aTypes: Array[Type], rType: Type)(impl: (EmitRegion, Array[Code[_]]) => Code[_]) {
+  def registerCode(mname: String, aTypes: Array[Type], rType: Type)(impl: (EmitRegion, Array[(PType, Code[_])]) => Code[_]) {
     IRFunctionRegistry.addIRFunction(new IRFunctionWithoutMissingness {
       override val name: String = mname
 
       override val argTypes: Seq[Type] = aTypes
 
       override val returnType: Type = rType
 
-      override def apply(r: EmitRegion, args: Code[_]*): Code[_] = impl(r, args.toArray)
+      override def apply(r: EmitRegion, args: (PType, Code[_])*): Code[_] = impl(r, args.toArray)
     })
   }
 
-  def registerCodeWithMissingness(mname: String, aTypes: Array[Type], rType: Type)(impl: (EmitRegion, Array[EmitTriplet]) => EmitTriplet) {
+  def registerCodeWithMissingness(mname: String, aTypes: Array[Type], rType: Type)(impl: (EmitRegion, Array[(PType, EmitTriplet)]) => EmitTriplet) {
```

Comment:
As an aside, I find it a little strange that there are 2 ways of defining functions, one with implicit and one with explicit Unit. 1 way seems sufficient.

We seem to use both, which makes them seem different.

Reply:
Which ones? You mean one that uses `Code[T]` and one that uses `EmitTriplet`? The latter is a lower-level interface, and is used when we need to also deal with missing bits.

Reply:
I just mean

def fn() {

}

vs 

def fn(): Unit {

}

Reply:
Oh! We prefer the latter syntax.

Reply:
So should we take the opportunity to make this:

```scala
def registerCodeWithMissingness(mname: String, aTypes: Array[Type], rType: Type)(impl: (EmitRegion, Array[(PType, EmitTriplet)]) => EmitTriplet): Unit = {
```